### ISSUE TITLE
Horizontal bar chart legend padding & sort order regression in 4.0.2rc2

### ISSUE THREAD

**Issue by sfirke** [apache/superset#29404]
### Bug description

Testing out https://github.com/apache/superset/pull/29345 and it introduces a regression on a chart vs. 4.0.0.  The top bar of my chart now goes over the legend and its axis label is not visible.

It also seems to have reversed the sorting order.

### How to reproduce the bug

Create bar chart and switch on horizontal x-axis.  If there are other aspects of my chart that are relevant to the bug I am happy to hop on a Zoom or chat to try to isolate that.

### Screenshots/recordings

Before (4.0.0):
![image](https://github.com/apache/superset/assets/7569808/e4b50f6e-976f-4c49-82f6-228d4df1a87f)

4.0.2rc2:
![image](https://github.com/apache/superset/assets/7569808/c33a574d-8d3f-4d3d-bf4f-aab3077ab075)

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [ ] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [ ] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [ ] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by michael-s-molina** [comment]
> It also seems to have reversed the sorting order.

@sfirke this is expected as described in https://github.com/apache/superset/pull/29345:

> This will change the default y-axis order of horizontal bar charts.

About the problem with the bar that goes over the legend, it looks like it happens in some specific combination of the number of bars and labels length. It's an ECharts problem so we would need a fix there to consistently fix this problem for all charts. @villebro any idea?

Regarding Superset releases, I think the current state is still better than the state before the reverts. You can check the screenshots of https://github.com/apache/superset/pull/29345 to see the problems.

So, my vote would be to proceed with the release and open a bug in ECharts for this problem.

**Comment by sfirke** [comment]
Wow, I had added a legend margin of -75 in my test instance from 4.0.2rc1 when I was seeing how I would work around this. Then I didn't reset the metadata DB for testing RC2.  This is just fine now.

And re: the sorting, that's fine with me, I had missed that sentence in the PR.  I just unchecked the box `Y-Axis Sort Ascending` and I get back to my original order under 4.0.0.  That's easy enough.

Sorry for the false alarm issue, thank you for the quick reply and investigation.  I will close this.
